src/Entity/Carrinho.php

```diff
@@ -123,7 +121,7 @@ public function getAtualizadoEm(): ?\DateTimeInterface
         return $this->atualizadoEm;
     }
 
-    public function setAtualizadoEm(\DateTimeInterface $atualizadoEm): static
+    public function setAtualizadoEm(?\DateTimeInterface $atualizadoEm): static
```

Comment:
Evita usar as interfaces sempre que saber o parametro... muda os outros lugares!
```suggestion
    public function setAtualizadoEm(?\DateTimeImmutable $atualizadoEm): static
```